### PR TITLE
rpctest: use errgroup to catch goroutine errors

### PR DESCRIPTION
**Pull request by dajohi**

### CODE REVIEW

**Comment by dajohi** [comment]
```
rpctest/rpc_harness_test.go:371:2: SA2002: the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (staticcheck)
	go func() {
	^
rpctest/rpc_harness_test.go:393:2: SA2002: the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (staticcheck)
	go func() {
	^
rpctest/rpc_harness_test.go:446:2: SA2002: the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (staticcheck)
	go func() {
	^
```

File: run_tests.sh [hunk 1]
```diff
@@ -18,17 +18,12 @@ REPO=dcrd
 
 go version
 
-# binary needed for RPC tests
-env CC=gcc go build
-cp "$REPO" "$(go env GOPATH)/bin/"
```
Comment (davecgh) [inline run_tests.sh]: Why is this being removed?  It is needed to ensure the latest dcrd is being used by the RPC tests in all environments.

File: run_tests.sh [hunk 2]
```diff
@@ -18,17 +18,12 @@ REPO=dcrd
...
-
 # run tests on all modules
 ROOTPATH=$(go list -m)
 ROOTPATHPATTERN=$(echo $ROOTPATH | sed 's/\\/\\\\/g' | sed 's/\//\\\//g')
 MODPATHS=$(go list -m all | grep "^$ROOTPATHPATTERN" | cut -d' ' -f1)
 for module in $MODPATHS; do
   echo "==> ${module}"
-  env CC=gcc go test -short -tags rpctest ${module}/...
```
Comment (davecgh) [inline run_tests.sh]: This would remove the tests.

File: run_tests.sh [hunk 3]
```diff
@@ -39,12 +34,7 @@ for module in $MODPATHS; do
   (cd $MODNAME && \
     go mod download && \
     golangci-lint run --build-tags=rpctest --disable-all --deadline=10m \
-      --enable=gofmt \
```
Comment (davecgh) [inline run_tests.sh]: huh?

Comment (dajohi) [reply run_tests.sh]: not supposed to be committed.

**Pull request merged**